packages/data/src/components/with-dispatch/index.js

```diff
@@ -27,63 +27,57 @@ import { RegistryConsumer } from '../registry-provider';
  * @return {Component} Enhanced component with merged dispatcher props.
  */
 const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent(
-	compose( [
-		pure,
```

Comment:
Any reason to remove pure?

Reply:
> Any reason to remove pure?

The thought was since we no longer respond to props changes, nor perform any expensive operations in the render, they could just go straight through.

The main downside I might see is if someone had come to rely on `withDispatch` for its pure behavior, which is and always would be a fragile dependency.